src/dtoannotations/Validator.java

```diff
@@ -0,0 +1,70 @@
+package dtoannotations;
+
+import java.lang.reflect.Field;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Validator {
+
+    public List<String> validate(Object object) throws IllegalAccessException {
+        List<String> list = new ArrayList<>();
+        Class<?> objectClass = object.getClass();
+        Field[] declaredFields = objectClass.getDeclaredFields();
+        for (Field declaredField : declaredFields) {
+            Object fieldValue = declaredField.get(objectClass);
```

Comment:
You have to use **object** instead **objectClass**, inside **get();** method, otherwise you will get IllegalAccessException. This is the reason, why you set CustomerDTO's fields **static**.

Reply:
And also you have to **CAST** it into the necessary type.
For example here ->. (String) declaredField.get(object);

Reply:
first part is done. Casting to String is redundant here (I tried, its default type is String), where necessary I did casting (cast to int, cast to LocalTime).

Reply:
You can easily **cast** all fields inside the IF statement.